packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -76,25 +102,49 @@ export const useStoreSwitcher = props => {
         );
     }, [storeConfigData, availableStoresData]);
 
+    // Get suffix based on page type
+    const getSuffix = useCallback(
+        (storeCode, currentSuffix) => {
+            let suffix = currentSuffix;
+
+            if (pageType === 'CATEGORY') {
+                suffix =
+                    availableStores.get(storeCode).category_url_suffix || '';
+            }
+            if (pageType === 'PRODUCT') {
+                suffix =
+                    availableStores.get(storeCode).product_url_suffix || '';
+            }
+
+            return suffix;
+        },
+        [availableStores, pageType]
+    );
+
     const handleSwitchStore = useCallback(
         // Change store view code and currency to be used in Apollo link request headers
         storeCode => {
             // Do nothing when store view is not present in available stores
             if (!availableStores.has(storeCode)) return;
 
+            // Use window.location.pathname to get the path with the store view code
+            // pathname from useLocation() does not include the store view code
+            const [pathName, index] = window.location.pathname.split('.');
```

Comment:
I think we can use current suffix for page type from config of the current store.

Reply:
For pages that are not category/product, such as `search.html` we may need to derive the suffix from window.location.pathname. That case is a client-side route, and pageType returned is `undefined`.

Reply:
I don't think we need to derive suffix from pathname from the page types other that category/product, just use the whole pathname as is.